web/modules/custom/portland/portland.module

```diff
@@ -585,22 +585,29 @@ function portland_search_api_solr_query_alter(\Solarium\QueryType\Select\Query\Q
     return;
   }
 
-  // Must get the original keys
   $keys = trim($query->getKeys());
-  // If keys only contains "@", assume it means "at"
-  if($keys === "@") $keys = "at";
-  // Convert "OR" at the start or end to lower case
-  $keys = preg_replace("/(\bOR$|^OR\b)/", "or", $keys);
-  // Convert "AND" at the start or end to lower case
-  $keys = preg_replace("/(\bAND$|^AND\b)/", "and", $keys);
-  // Remove whole word "||" at the start and end
-  $keys = preg_replace("/(\b\|\|$|^\|\|\b)/", "", $keys);
-  // "+" and "-" can't be at the end as whole word
-  $keys = preg_replace("/\b(\+|\-)$/", "", $keys);
-  // "^" has to be appended immediately before a number. Also can't be the only character
-  $keys = preg_replace("/\^([^0-9\.]+|$)/", "", $keys);
-  // If keywords only contain "&", or if it's exactly "||", remove it
-  $keys = preg_replace("/(\&+|^\|\|$)/", "", $keys);
+  $keys = preg_replace([
+    // Convert "OR" and "||" at the start or end to "or"
+    "/^OR\b|\bOR$/",
+    "/^\|\|(?!\w)|(?!\w)\|\|$/",
+    // Convert "AND" and "&&" at the start or end to "and"
+    "/^AND\b|\bAND$/",
+    "/^\&\&(?!\w)|(?!\w)\&\&$/",
+    // "+" and "-" can't be at the end with space in any combination
+    "/\s[\+\- \"]+$/",
+    // "^" has to be appended immediately before a number
+    "/\^[^0-9\.]+/",
```

Comment:
FYI I still get an error if I search for "^1", "^0.1", or "^.1", although admittedly it's highly unlikely a user will search for those strings.

Reply:
Ah, good catch! Added a pattern to detect that.